traffic_portal/app/src/modules/private/cacheStats/CacheStatsController.js

```diff
@@ -17,60 +17,218 @@
  * under the License.
  */
 
-var CacheStatsController = function(cacheStats, $scope, $state, $interval, numberUtils, serverUtils) {
+/**
+ * @typedef CacheStat
+ * @property {string} cachegroup
+ * @property {number} connections
+ * @property {boolean} healthy
+ * @property {string} hostname
+ * @property {string | null} ip
+ * @property {number} kbps
+ * @property {string} profile
+ * @property {string} status
+ */
 
-	var cacheStatsInterval,
-		autoRefresh = false,
-		refreshRateInMS = 10000;
 
-	var createInterval = function() {
-		killInterval();
-		cacheStatsInterval = $interval(function() { $scope.refresh() }, refreshRateInMS );
-	};
+class CacheStatsHealthyCellRenderer {
```

Comment:
why is this class not nested but `CacheStatsSSHCellRenderer` is?

Reply:
Because `CacheStatsSSHCellRenderer` needs access to the injected `userModel` service, but `CacheStatsHealthyCellRenderer` is pure functional